src/arithmetic.jl

```diff
@@ -0,0 +1,501 @@
+# This file is part of the TaylorSeries.jl Julia package, MIT license
+#
+# Luis Benet & David P. Sanders
+# UNAM
+#
+# MIT Expat license
+#
+
+# Arithmetic operations: +, -, *, /
+
+## Equality ##
+for T in (:Taylor1, :TaylorN)
+    @eval begin
+        =={T<:Number,S<:Number}(a::$T{T}, b::$T{S}) = ==(promote(a,b)...)
+        function =={T<:Number}(a::$T{T}, b::$T{T})
+            a, b = fixorder(a, b)
+            @simd for i in eachindex(a.coeffs)
+                @inbounds (a.coeffs[i] == b.coeffs[i]) || return false
```

Comment:
I would guess it would be the same speed and much less writing. You are just writing out by hand testing equality of two vectors as far as I can tell.

If you don't  use `fixorder` then testing equality would indeed be more complicated.